ui/components/app/import-token/network-filter-import-token/index.tsx

```diff
@@ -0,0 +1,214 @@
+
+  const renderItem = () => {
+    if (isCurrentNetwork) {
+      return (
+        <Box
+          display={Display.Flex}
+          justifyContent={JustifyContent.spaceBetween}
+          width={BlockSize.Full}
+          paddingTop={3}
+          paddingBottom={3}
+          gap={3}
+        >
+          <Box>
+            <Text
+              variant={TextVariant.bodyMdMedium}
+              color={TextColor.textDefault}
+            >
+              {t('currentNetwork')}
+            </Text>
+          </Box>
+          <Box
+            display={Display.Flex}
+            alignItems={AlignItems.flexStart}
+            onClick={openListNetwork}
+          >
+            <AvatarNetwork
+              key={currentNetworkImageUrl}
+              name={currentNetworkImageUrl ?? ''}
+              src={currentNetworkImageUrl ?? undefined}
+              size={AvatarNetworkSize.Sm}
+            />
+          </Box>
+        </Box>
+      );
+    }
+    return (
+      <Box
+        display={Display.Flex}
+        justifyContent={JustifyContent.spaceBetween}
+        width={BlockSize.Full}
+        paddingTop={3}
+        paddingBottom={3}
+        gap={3}
+      >
+        <Text variant={TextVariant.bodyMdMedium} color={TextColor.textDefault}>
+          {t('popularNetworks')}
+        </Text>
+        <Box
+          display={Display.Flex}
+          alignItems={AlignItems.flexEnd}
+          onClick={openListNetwork}
+        >
+          {FEATURED_NETWORK_CHAIN_IDS.filter((chain) => allOpts[chain]).map(
+            (chain, index) => {
+              const networkImageUrl =
+                CHAIN_ID_TO_NETWORK_IMAGE_URL_MAP[
+                  chain as keyof typeof CHAIN_ID_TO_NETWORK_IMAGE_URL_MAP
+                ];
+              return (
+                <AvatarNetwork
+                  key={networkImageUrl}
+                  name={networkImageUrl}
+                  src={networkImageUrl ?? undefined}
+                  size={AvatarNetworkSize.Sm}
+                  style={{
+                    marginLeft: index === 0 ? 0 : '-20px',
+                    zIndex: 5 - index,
+                  }}
+                />
+              );
+            },
+          )}
+        </Box>
+      </Box>
+    );
+  };
+
+  const box = (
+    <Box
+      className="dropdown-editor__item-dropdown"
+      display={Display.Flex}
+      alignItems={AlignItems.center}
+      justifyContent={JustifyContent.spaceBetween}
+      borderRadius={BorderRadius.LG}
+      borderColor={BorderColor.borderDefault}
+      borderWidth={1}
+      paddingLeft={4}
+      paddingRight={1}
+      ref={dropdown}
+    >
+      {renderItem()}
+      <ButtonIcon
+        marginLeft="auto"
+        iconName={isDropdownOpen ? IconName.ArrowUp : IconName.ArrowDown}
+        ariaLabel={title}
+        size={ButtonIconSize.Md}
+        onClick={() => {
+          setIsDropdownOpen(!isDropdownOpen);
+        }}
+        data-testid={buttonDataTestId}
+      />
+    </Box>
+  );
```

Comment:
Hmm, my preference here would be to create separate components for `box` and `renderItem` here, and just pass them the props that they need, rather that using internal callbacks here to conditionally render jsx. I find that this pattern leads to leaner components that are easier to troubleshoot.

Was there a reason for taking this approach? Outside of readability, I worry that because these callback are happening in the component body, and aren't memoized, that they may cause unnecessary rerenders.